addons/loyalty/models/loyalty_reward.py

```diff
@@ -105,6 +106,10 @@ def _compute_reward_product_uom_id(self):
         for reward in self:
             reward.reward_product_uom_id = reward.reward_product_ids.product_tmpl_id.uom_id[:1]
 
+    def _search_reward_product_ids(self, operator, value):
+        assert operator in ('=', '!=', 'in')
+        return ['|', ('reward_product_id', operator, value), ('reward_product_tag_id.product_ids', operator, value)]
```

Comment:
seems ok but if we want to match the compute function, it seems that we should have

```suggestion
        return ['&', ('reward_type', '=', 'product'), '|', ('reward_product_id', operator, value), ('reward_product_tag_id.product_ids', operator, value)]
```